src/editors/editorsController.ts

```diff
@@ -229,117 +232,130 @@ export default class EditorsController {
+    namespace: string,
+    dataservice: DataServiceType,
+    editor: vscode.TextEditor
+  ): Promise<boolean> {
+    let newDocument: EJSON.SerializableTypes = {};
 
-      log.info('save document to MongoDB', activeEditor);
+    try {
+      newDocument = EJSON.parse(editor.document.getText());
+    } catch (error) {
+      // Send metrics to Segment.
+      this._telemetryController.trackDocumentUpdated('treeview', false);
 
-      if (!activeEditor) {
-        return resolve(true);
-      }
+      vscode.window.showErrorMessage(error.message);
 
-      const uriParams = new URLSearchParams(activeEditor.document.uri.query);
-      const documentLocation =
-        uriParams.get(DOCUMENT_LOCATION_URI_IDENTIFIER) || '';
-      const namespace = uriParams.get(NAMESPACE_URI_IDENTIFIER) || '';
-      const connectionId = uriParams.get(CONNECTION_ID_URI_IDENTIFIER);
-      const documentIdReference =
-        uriParams.get(DOCUMENT_ID_URI_IDENTIFIER) || '';
-      const documentId = this._documentIdStore.get(documentIdReference);
-
-      if (
-        documentLocation !== 'mongodb' ||
-        !namespace ||
-        !connectionId ||
-        !documentId
-      ) {
-        vscode.commands.executeCommand('workbench.action.files.save');
+      return false;
+    }
 
-        return resolve(true);
-      }
+    const findOneAndReplace = util.promisify(dataservice.findOneAndReplace);
 
-      const activeConnectionId = this._connectionController.getActiveConnectionId();
-      const connectionName = this._connectionController.getSavedConnectionName(
-        connectionId
+    try {
+      await findOneAndReplace(
+        namespace,
+        {
+          _id: documentId
+        },
+        newDocument,
+        {
+          returnOriginal: false
+        }
       );
 
-      if (activeConnectionId !== connectionId) {
-        // Send metrics to Segment.
-        this._telemetryController.trackDocumentUpdated('treeview', false);
+      this._statusView.hideMessage();
 
-        vscode.window.showErrorMessage(
-          `Unable to save document: no longer connected to '${connectionName}'`
-        );
+      // Send metrics to Segment.
+      this._telemetryController.trackDocumentUpdated('treeview', true);
 
-        return resolve(false);
-      }
+      editor.document.save();
+      vscode.window.showInformationMessage(
+        `The document was saved successfully to '${namespace}'`
+      );
 
-      const dataservice = this._connectionController.getActiveDataService();
+      return true;
+    } catch (error) {
+      this._statusView.hideMessage();
 
-      if (dataservice === null) {
-        // Send metrics to Segment.
-        this._telemetryController.trackDocumentUpdated('treeview', false);
+      const errorMessage = `Unable to save document: ${error.message}`;
 
-        vscode.window.showErrorMessage(
-          `Unable to save document: no longer connected to '${connectionName}'`
-        );
+      // Send metrics to Segment.
```

Comment:
this code is duplicated here and there, what about:

``` js
failSaveDocument(errorMessage) {
   this._telemetryController.trackDocumentUpdated('treeview', false);
        );	


        return resolve(false);	      vscode.window.showErrorMessage(errorMessage);
}
```

Reply:
Good call. Updated & updated the comment to be a bit more generic.